### DISCUSSION TITLE
Reuse inactive session

### DISCUSSION THREAD

**Discussion by alexluigit** [alexluigit/dirvish#57] [Show and tell]
After cc0c04f0aaf0bc9649bf0f645b11f3b5741a28f8, users should be able to reuse a hidden session (if there is any) by `dired-jump` or `dirvish-dwim` when `dirvish-keep-alive-on-quit` is set to t. It allows us to keep only one dirvish buffer in the buffer list when there are no visible dirvish sessions (exception: `dirvish-side` can still hold one or more sessions if being used).

The way that dirvish manages its buffers is documented [here](https://github.com/alexluigit/dirvish/blob/main/CUSTOMIZING.org#buffer-management).

In short, 
1. Dirvish reuses its session-local buffers on directory navigation rather than kill them on the fly like `dired-kill-when-opening-new-dired-buffer` does.
2. Dirvish cleans its buffers on session quit (triggered by `find-file` or `quit-window`)

Doing so we get best things from both worlds: fully-reused buffers (essential for big/tramp directories) and a clean buffer list.

Feel free to post any thoughts on this or Dirivish's buffer management in general. Thanks.

**Comment by aikrahguzar** [comment]
Hi @alexluigit 
  Thanks a lot for this change. It makes it much better fit for how I use Dired. But is it possible to make buffer management by Dirvish more minimal? Ideally I would like it to make Dirvish buffers I create by `find-file` to stay unless I navigate to one of their sub-directories or kill them. Now `dired-jump` behaves as expected but using `find-file` results in one of the already open buffers disappearing.

P.S. Is there a way to disable header line in Dirvish?

**Comment by alexluigit** [comment]
> but using find-file results in one of the already open buffers disappearing

What do you mean by this?
Assume I'm editing /foo/123.txt
1. dired-jump (gives me /foo)
2. quit-window
3. find-file "/bar/" (gives me /bar)
4. did you mean "/foo" get killed here?

I can not reproduce this on emacs -q, and I did not add any session reuse logic to `find-file` as of this update. 

> Is there a way to disable header line in Dirvish?

```emacs-lisp
(setq dirvish-header-line-position 'disable)
```
This only work for fullscreen session for now. Maybe I'll let this option take effect for single window session as well.

**Reply by aikrahguzar** [reply]
> What do you mean by this? Assume I'm editing /foo/123.txt
> 
>     1. dired-jump (gives me /foo)
> 
>     2. quit-window
> 
>     3. find-file "/bar/" (gives me /bar)
> 
>     4. did you mean "/foo" get killed here?

Essentially this but it is somewhat intermittent. I will try to get a reproducible example and report back.


> This only work for fullscreen session for now. Maybe I'll let this option take effect for single window session as well.

Please do, I think that will be good. Or maybe just allow allow a special symbol like `always` to make it take effect for single windows if you think different behavior for fullscreen session is an important use case.
